diff --git a/dev/archery/archery/integration/datagen.py b/dev/archery/archery/integration/datagen.py
index 422130738..22e7e1834 100644
--- a/dev/archery/archery/integration/datagen.py
+++ b/dev/archery/archery/integration/datagen.py
@@ -1626,7 +1626,8 @@ def get_generated_json_files(tempdir=None):
         generate_non_canonical_map_case()
         .skip_category('C#')
         .skip_category('Java')   # TODO(ARROW-8715)
-        .skip_category('JS'),     # TODO(ARROW-8716)
+        .skip_category('JS')     # TODO(ARROW-8716)
+        .skip_category('Rust'),   # TODO(ARROW-8715)
 
         generate_nested_case()
         .skip_category('C#'),
